cl_manycore/regression/cuda/Makefile

```diff
@@ -0,0 +1,12 @@
+include Makefile.tests
+include ../../Makefile.dimensions
+include ../Makefile.common
+
+$(TARGETS): test_%: $(BSG_MANYCORE_DIR)/software/spmd/bsg_cuda_lite_runtime/%/binary
+
+$(BSG_MANYCORE_DIR)/software/spmd/bsg_cuda_lite_runtime/%/binary:
+	$(MAKE) -C $(dir $@) clean main.riscv \
+		bsg_tiles_X=2 \
```

Comment:
Can you use the dimensions parsed from Makefile.deps like we do in the spmd tests?

https://github.com/bespoke-silicon-group/bsg_f1/blob/dbf65db99b9032df52baa7debe6cbd0b2d3c2d22/cl_manycore/regression/spmd/Makefile#L9

Reply:
@drichmond So basically I had to change two things:
1. In the Makefile I set the dimensions to CL_MANYCORE_DIM_X/Y to be used for building the hardware with these  dimensions
2. I added TILE_GROUP_DIM_X/Y to the Makefile.dimensions, which are used in Makefile.test for compiling the riscv binary. Because the riscv dimensions should not be same as the entire mesh dimensions anymore, they should match the tile group dimensions otherwise barrier won't work, as Max found out a while ago. 
What do you think?